index.js

```diff
@@ -141,44 +173,64 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+  var ticketNumber = 1
+  function takeATicketNumber(line) {
+    line.push(ticketNumber)
+    var newMsg = "Welcome. You are ticket number " + ticketNumber
+      ticketNumber++;
+      return newMsg;
+  }
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
+
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+ ticketNumber = 1
+// function ticketNumberGeneratorFunc(){
+//   return takeATicketNumber(newLine)
+//   newLine.push(ticketNumberNew); 
+//   var message = "Welcome. You are ticket number " + ticketNumberNew
+//   ticketNumberNew++;
+//   return message
+// }
+console.log("%cScope & Closures - Question 2", "color: red")
+
+const newLine = []
+const takeATicketNumberFunc = function ticketNumberGeneratorFunc(newLine){
+  return takeATicketNumber(newLine);
+}
```

Comment:
Were you able to figure this out since? Let me know if you want to work through this together!